irohad/ametsuchi/tx_presence_cache.hpp

```diff
@@ -0,0 +1,53 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef IROHA_TX_PRESENCE_CACHE_HPP
+#define IROHA_TX_PRESENCE_CACHE_HPP
+
+#include <vector>
+
+#include "ametsuchi/tx_cache_outcome.hpp"
+
+namespace shared_model {
+  namespace crypto {
+    class Hash;
+  }  // namespace crypto
+
+  namespace interface {
+    class TransactionBatch;
+  }  // namespace interface
+}  // namespace shared_model
+
+namespace iroha {
+  namespace ametsuchi {
+
+    /**
+     * Class is responsible for checking transaction status in the ledger
+     */
+    class TxPresenceCache {
+     public:
+      /**
+       * Check transaction status by hash
+       * @param hash - corresponding hash of the interested transaction
```

Comment:
```suggestion
       * @param hash - hash to be checked
```
I think a single-line docstring is sufficient here.